docs/git-bundles.md

```diff
@@ -21,8 +21,86 @@ When creating a Bundle from a git source, a reference to a particular commit, ta
 provisioner to know where the bundle content is stored in the repository. Only one can be specified, and it is expected
 that the manifests are present in the particular commit/tag/branch at the directory specified.
 
-Currently only publicly-available git repositories are available as backing sources to Bundles. Authentication and using
-private git repositories as Bundle sources is on the roadmap.
+## Private git repositories
+
+A Bundle can reference content in a private git repository by creating a secret in the namespace that the provisioner is deployed.
+For `https` URL, the secret is a [Basic authentication secret](https://kubernetes.io/docs/concepts/configuration/secret/#basic-authentication-secret)
+and expected to contain `data.username` and `data.password` for the username and password, respectively.
+For `ssh` URL, the secret is a [SSH authentication secrets](https://kubernetes.io/docs/concepts/configuration/secret/#ssh-authentication-secrets)
+and expected to contain `data.ssh-privatekey` and `data.ssh-knownhosts` for the ssh privatekey and the host entry in the known_hosts file respectively.
+If the `git.auth.insecureSkipVerify` is set true, the clone operation will accept any certificate presented by the server and any host name in that
+certificate. In this mode, TLS is susceptible to machine-in-the-middle attacks unless custom verification is
+used. This should be used only for testing.
+
+### Example steps for `https` URL
+
+1. Create the secret
+
+```sh
+echo -n 'user name' > username.txt
+echo -n 'password' > password.txt
+kubectl create secret generic gitsecret --type "kubernetes.io/basic-auth" --from-file=username=./username.txt --from-file=password=./password.txt -n rukpak-system
+```
+
+2. Create a private repository (private-registry/combo) and copy operator-framework/combo contents into it
```

Comment:
nit: I'm not sure the user should have to create a new private repository and copy combo in order to show how private git repo access works. I would maybe just say something like
`Find an existing private git repository or create a new one that is private by default`

Reply:
It seems better.  I'll change it.  Thanks!